elfpy/utils/apeworx_integrations.py

```diff
@@ -469,6 +470,64 @@ def get_agent_deltas(tx_receipt: ReceiptAPI, trade, addresses, trade_type, pool_
     return agent_deltas
 
 
+def get_instance(address: str, provider: ProviderAPI) -> ContractInstance:
+    """Instantiate Contract at a specific address, explicitly using the cache (where Ape refuses to).
+
+    Parameters
+    ----------
+    address : str
+        Address of the contract to instantiate.
+    provider : ` ape.api.providers.ProviderAPI <https://docs.apeworx.io/ape/stable/methoddocs/api.html#ape.api.providers.ProviderAPI>`_
+        Ape Provider object represents a connection to a blockchain network.
+
+    Example
+    -------
+    >>> faucet = ape_utils.get_instance(FAUCET_ADDRESS, provider=provider)
+
+    Returns
+    -------
+    ContractInstance
+        Contract instance at the specified address.
+    """
+    contract_type = get_contract_type(address, provider=provider)
+    return Contract(address=address, contract_type=contract_type)
+
+
+def get_contract_type(address: str, provider: ProviderAPI) -> ContractType:
+    """Get contract type from cache. Used for devnet, where Ape refuses to check the cache
+
+    Parameters
+    ----------
+    address : str
+        Address of the contract to instantiate.
+    provider : ` ape.api.providers.ProviderAPI <https://docs.apeworx.io/ape/stable/methoddocs/api.html#ape.api.providers.ProviderAPI>`_
+        Ape Provider object represents a connection to a blockchain network.
+
+    Example
+    -------
+    >>> faucet = Contract(FAUCET_ADDRESS, contract_type=ape_utils.get_contract_type(FAUCET_ADDRESS, provider=provider))
+
+    Returns
+    -------
+    ContractType
+        Contract type at the specified address.
+    """
+    # pylint: disable=protected-access
+    # sourcery skip: use-named-expression
```

Comment:
We haven't (and probably wont) adopted sourcery. I guess it doesn't hurt to leave this one instance, but if we end up needing to ignore a lot then I'd recommend instead setting up your own local config file that ignores stuff.